### PR TITLE
Add few world methods

### PR DESCRIPTION
**Pull request by SplotyCode**
# Description

# Checklist
The following items should be checked before the pull request can be merged.
- [X] Code follows existing style.
- [X] Unit tests added (if applicable).

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/WorldMock.java [hunk 1]
```diff
@@ -2817,29 +2819,25 @@ public boolean isRespawnAnchorWorks()
 	@Override
 	public boolean hasRaids()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return environment != Environment.NETHER;
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/WorldMock.java:2822]: Does the end have raids?

File: src/main/java/be/seeseemelk/mockbukkit/WorldMock.java [hunk 2]
```diff
@@ -511,10 +510,9 @@ public boolean hasCollisionsIn(@NotNull BoundingBox boundingBox)
 
 	@Override
 	@Deprecated
-	public @NotNull Block getBlockAtKey(long key)
+	public @NotNull BlockMock getBlockAtKey(long key)
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/WorldMock.java]: Is there a reason the return type here was changed?

**Review by Insprill — CHANGES_REQUESTED** [review]
Please fix your commit signature as well.

File: src/main/java/be/seeseemelk/mockbukkit/WorldMock.java (hunk 2 above)
Comment (SplotyCode) [reply src/main/java/be/seeseemelk/mockbukkit/WorldMock.java]: Just wanted to align with the other block methods we currently provide. Methods like `assertType` on BlockMock could come handy in tests. I really dont mind and will let you decide on witch fits the best

Comment (Insprill) [reply src/main/java/be/seeseemelk/mockbukkit/WorldMock.java]: I think we should cross that bridge when we get there, no need to change it now without a current reason.

File: src/main/java/be/seeseemelk/mockbukkit/WorldMock.java (hunk 1 above)
Comment (SplotyCode) [reply src/main/java/be/seeseemelk/mockbukkit/WorldMock.java:2822]: Havent looked at it ingame. Just wanted to replicate paper logic witch uses nms getter.
net.minecraft.data.worldgen.DimensionTypes
Bildschirmfoto 2023-10-11 um 19 19 27

( line wraps were added for better readbilityy )

File: src/main/java/be/seeseemelk/mockbukkit/WorldMock.java (hunk 2 above)
Comment (SplotyCode) [reply src/main/java/be/seeseemelk/mockbukkit/WorldMock.java]: sure lets keep the old signature 👍

**Comment by Insprill** [comment]
Please fix your GPG signature and resign the commits before we merge this.

**Comment by SplotyCode** [comment]
Fixed

**Review by Insprill — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**